### ISSUE TITLE
Ordering of namelist

### ISSUE THREAD

**Issue by dabail10** [CICE-Consortium/CICE#134]
I know this should not be an issue, but on hobart with the NAG compiler, the order of the namelists in ice_in matters. As long as it is not a big deal, I would like to have these in the order that they are read in in ice_init.F90.

Dave

**Comment by eclare108213** [comment]
Which ones are out of order?
DMI has suggested breaking them out into individual files.

**Comment by dabail10** [comment]
The order in ice_init.F90 is:

setup_nml
grid_nml
tracer_nml
thermo_nml
dynamics_nml
shortwave_nml
ponds_nml
forcing_nml

So domain_nml would move down to just before bgc_nml. forcing_nml would move up to just after ponds_nml and before domain_nml. Technically, we could introduce a "rewind" for each of these reads, but this reordering also works.

**Comment by apcraig** [comment]
I would make an argument that we do not want to split them into separate files and it shouldn't matter what order they appear.  If nag is complaining, then is there something we can do in the implementation to make it stop complaining.  There should not be a requirement that the namelists appear in a certain order and we should not test that they are in a certain order.

If nag is throwing bunches of warnings because of this, and there is nothing we can do to turn that check off or to work around it with some code changes, then I am willing to change the namelist order for nag somehow.  But I think this should be a last resort, and that maybe we even should just let nag complain.

**Comment by dabail10** [comment]
No, NAG simply aborts because it cannot read the namelist in the current order. A way to fix it more generally is to put a "rewind" in there after each read statement.

Dave

**Comment by apcraig** [comment]
OK, I assumed we had a rewind in the implementation.  If not, we should add that.

**Comment by eclare108213** [comment]
@dabail10 you've recently rearranged the namelists in #153.  Did that take care of this issue?
